faultfree:
  P1: 
    # Q: [900, 900, 150, 10, 10, 10]
    # Q: [300, 300, 150, 10, 10, 10]
    # Q: [5, 5, 10, 0.2, 0.2, 1000]
    # Q: [10, 10, 100, 10, 10, 100]
    # Q: [1, 1, 100, 0.1, 0.1, 0.1]
    # Q: [100, 100, 100, 10, 10, 0.1]
    Q: [10, 10, 10, 1, 1, 0.1]
    # R: [0.1, 0.1, 0.1]
    R: [1, 1, 1]
    R_full: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
    # R_full: [1, 1, 1, 1, 1, 1, 1, 1]
    P_mult: 1  # P = P_mult * Q
    terminal_const_file: "controllers/Tuning_Helpers/terminal_incredients_faultfree.npz"
  # Backup
  P2: 
    Q: [5, 5, 10, 0.2, 0.2, 0.1]
    R: [1, 1, 1]
    R_full: [1, 1, 1, 1, 1, 1, 1, 1]
    P_mult: 1  # P = P_mult * Q
spiraling:
  P1:
    Q: [15, 15, 0.2, 0.2]
    # Q: [5, 5, 0.2, 0.2]
    # R: [0.1, 0.1, 0.1]
    R: [0.001, 0.001, 0.001]
    # R: [1, 1, 1]
    # R_full: [0.1, 0.1, 0.1]
    # R_full: [1, 1, 1]
    P_mult: 1
    poles: [-1, -1.25, -1.5, -1.75, -0.5, -0.75] # poles for the feedback-linearized system
spiraling_5:
  P1:
    # Q: [5, 5, 0.2, 0.2, 0.1]
    Q: [5, 5, 0.2, 0.2, 100]
    # R: [100, 100, 100]
    R: [1, 1, 1]
    # R: [0, 0, 0]
    # R: [0.001, 0.001, 0.001]
    P_mult: 1
    # poles: [-1, -1.25, -1.5, -1.75, -0.5, -0.75] # poles for the feedback-linearized system
    Q_linfb: [1, 1, 1, 1, 1]
    R_linfb: [10, 10, 10]
explicit_mpc_spiraling:
  P1:
    Q: [1, 1, 1, 1, 0]
    # Q: [1, 1, 1, 1, 1]
    # Q: [5, 5, 0.2, 0.2, 0.1]
    # R: [0.1, 0.1, 0.1]
    R: [0.1, 0.1, 0.01]
    P_mult: 1
    Q_linfb: [1, 1, 1, 1, 1]
    R_linfb: [10, 10, 10]
    k_omega: 0.1
    max_acceleration: 0.0
fb_lin_5:
  P1:
    Q: [5, 5, 0.2, 0.2, 100]
    R: [1, 1, 1]
    P_mult: 1
    Q_linfb: [0.1, 0.1, 1, 1, 1]
    R_linfb: [10, 10, 10]
globals:
  dt: 0.1